lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1687,31 +1703,34 @@ private function generateIdentifierName($identifier)
 
     protected function getCommentOnTableSQL(string $tableName, ?string $comment): string
     {
-        return sprintf(
-            <<<'SQL'
-EXEC sys.sp_addextendedproperty @name=N'MS_Description',
-  @value=N%s, @level0type=N'SCHEMA', @level0name=N'dbo',
-  @level1type=N'TABLE', @level1name=N%s
-SQL
-            ,
-            $this->quoteStringLiteral((string) $comment),
-            $this->quoteStringLiteral($tableName)
+        if (strpos($tableName, '.') !== false) {
+            [$schemaName, $tableName] = explode('.', $tableName, 2);
+        } else {
+            $schemaName = $this->getDefaultSchemaName();
+        }
+
+        return $this->getAddExtendedPropertySQL(
+            'MS_Description',
+            (string) $comment,
+            'SCHEMA',
+            $schemaName,
+            'TABLE',
+            $tableName
         );
     }
 
     public function getListTableMetadataSQL(string $table): string
     {
         return sprintf(
             <<<'SQL'
-SELECT
-  p.value AS [table_comment]
-FROM
-  sys.tables AS tbl
-  INNER JOIN sys.extended_properties AS p ON p.major_id=tbl.object_id AND p.minor_id=0 AND p.class=1
-WHERE
-  (tbl.name=N%s and SCHEMA_NAME(tbl.schema_id)=N'dbo' and p.name=N'MS_Description')
-SQL
-            ,
+            SELECT
+              p.value AS [table_comment]
+            FROM
+              sys.tables AS tbl
+              INNER JOIN sys.extended_properties AS p ON p.major_id=tbl.object_id AND p.minor_id=0 AND p.class=1
+            WHERE
+              (tbl.name=N%s and SCHEMA_NAME(tbl.schema_id)=N'dbo' and p.name=N'MS_Description')
+            SQL,
```

Comment:
This change is welcome, but in a separate commit that explains why you do it. Are there other changes in this PR that are not related to the fix? If yes, please move them to a separate commit, it will make it easier for me to understand. Ideally, you should have 1 commit with only what is strictly necessary to fix the issue + the tests, and all the rest should be in separate commits.

Reply:
@greg0ire you are great guy, I am maintainer of several projects as well, ok, you have twice the yearly contributions I have, but please, if this is needed, please do it so. We are not inventing a rocket and as you can see in the functional test, Oracle and PostgreSQL platforms does not even meet the standard/abstract functionality (accept quoted identifier) - for these reasons, I left this for someone else to fix in a different/larger PR.

Also a reson, why I created https://github.com/doctrine/dbal/issues/4385 5 hours ago 🚀

Reply:
> if this is needed, please do it so.

This part is needed, but maybe there are other changes in this PR that are note related to the fix either? If not do you understand why I would like to separate those? It will make it easier for me to understand which part of the PR actually fixes the bug you say you are fixing. Again, use `git add -p`